python/tank/util/shotgun_entity.py

```diff
@@ -108,86 +110,135 @@ def __init__(self, tk, entity_type, field_name_expr):
         # We want them most inclusive(longest) version first
         self._sorted_exprs = sorted(expr_variations, key=lambda x: len(x), reverse=True)
 
-        # now extract and store a bunch of data for each variation.
+        # Extract and store a bunch of data for each variation.
         self._variations = {}
         for v in expr_variations:
             
             try:            
-                # find all field names ["xx", "yy", "zz.xx"] from "{xx}_{yy}_{zz.xx}"
-                fields = set(re.findall('{([^}^{]*)}', v))
+                # find all field names: "{code:^([^_]+)}_{yy}_{zz.xx}"
```

Comment:
It's unclear to me what this is doing. The old comment was a little clearer. Was this a find/replace thing? Maybe some more comments here to help people new to this code either way.

Reply:
fixed!